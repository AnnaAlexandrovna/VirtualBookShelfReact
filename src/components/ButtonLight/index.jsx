import './index.scss';
import { Routers } from '../../constants/routes';
import { Link } from 'react-router-dom';
import React from 'react';

const ButtonLight = React.memo(({ value, title }) =>
    <Link
        to={`${Routers.collectionWithoutId}${value}`} >
        <button
            title={title}
            className='button-light'
        >
            {title}
        </button>
    </Link>
);
export default ButtonLight;