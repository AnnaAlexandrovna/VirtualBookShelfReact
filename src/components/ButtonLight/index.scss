@import '../../styles/colors.scss';
@import '../../styles/displaySettings.scss';
@import '../../styles/boarderRadius.scss';
.button-light {
    background-color: $yellow-color-light;
    border-color: $grey-color-light;
    cursor: pointer;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    @include media("<=phone") {
        border: 0.1rem solid transparent;
        border-radius: $border-radius-less-phone;
        font-size: 60%;
        max-width: 7rem;
        padding: 1px 3px;
        margin: 0.1rem;
    }
    @include media(">phone") {
        border: 0.2rem solid transparent;
        border-radius: $border-radius-phone;
        max-width: 7rem;
        padding: 1px 3px;
        margin: 0.2rem;
    }
    @include media(">=815px", "<desktop") {
        border: 0.3rem solid transparent;
        border-radius: $border-radius-tablet;
        font-size: 90%;
        max-width: 9rem;
        padding: 1px 3px;
        margin: 0.3rem;
    }
    @include media(">=desktop") {
        border: 0.3rem solid transparent;
        border-radius: $border-radius-desktop;
        font-size: 90%;
        max-width: 9rem;
        padding: 1px 3px;
        margin: 0.3rem;
    }
    &:hover {
        background-color: $yellow-color-dark;
    }
}