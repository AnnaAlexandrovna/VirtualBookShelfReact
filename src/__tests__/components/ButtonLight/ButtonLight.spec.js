import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ButtonLight from '../../../components/ButtonLight';

Enzyme.configure({ adapter: new Adapter() });
let wrapper;
const createComponent = props => shallow(<ButtonLight {...props} />);
const propsInitial = { value: '123', title: '12334' };

describe('Render ButtonLight', () => {
    beforeAll(() => wrapper = createComponent(propsInitial));

    it('should have name "Link"', () => {
        expect(wrapper.name()).toBe('Link');
    });

    it('should have correct prop "to"', () => {
        expect(wrapper.props().to).toBe(`/collection/${propsInitial.value}`);
    });

    it('should have child with name "button"', () => {
        expect(wrapper.children().name()).toBe('button');
    });

    it('should have child with className "button-light"', () => {
        expect(wrapper.children().props().className).toBe('button-light');
    });

    it('should have child with title from props', () => {
        expect(wrapper.children().props().title).toBe(propsInitial.title);
    });

    it('should have child with value from props', () => {
        expect(wrapper.children().children().text()).toBe(propsInitial.title);
    });

    it('should render without props', () => {
        const wrapper = createComponent({});
        expect(wrapper.children().props().title).toBe(undefined);
        expect(wrapper.props().to).toBe('/collection/undefined');
        expect(wrapper.children().props().children).toBe(undefined);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });
});