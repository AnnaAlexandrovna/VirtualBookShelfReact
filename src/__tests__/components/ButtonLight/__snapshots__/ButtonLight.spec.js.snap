// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render ButtonLight should match snapshot 1`] = `
<Link
  to="/collection/123"
>
  <button
    className="button-light"
    title="12334"
  >
    12334
  </button>
</Link>
`;
